src/main/java/io/cdap/plugin/ariba/source/connector/AribaConnector.java

```diff
@@ -0,0 +1,160 @@
+import io.cdap.cdap.etl.api.connector.BrowseRequest;
+import io.cdap.cdap.etl.api.connector.Connector;
+import io.cdap.cdap.etl.api.connector.ConnectorContext;
+import io.cdap.cdap.etl.api.connector.ConnectorSpec;
+import io.cdap.cdap.etl.api.connector.ConnectorSpecRequest;
+import io.cdap.cdap.etl.api.connector.PluginSpec;
+import io.cdap.cdap.etl.api.validation.ValidationException;
+import io.cdap.plugin.ariba.source.AribaBatchSource;
+import io.cdap.plugin.ariba.source.AribaServices;
+import io.cdap.plugin.ariba.source.config.AribaPluginConfig;
+import io.cdap.plugin.ariba.source.exception.AribaException;
+import io.cdap.plugin.ariba.source.metadata.AribaResponseContainer;
+import io.cdap.plugin.ariba.source.util.ResourceConstants;
+import io.cdap.plugin.common.ConfigUtil;
+import okhttp3.HttpUrl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+
+/**
+ * AribaConnector Class
+ */
+@Plugin(type = Connector.PLUGIN_TYPE)
+@Name(ResourceConstants.PLUGIN_NAME)
+@Description("Connection to access data from Ariba.")
+
+public class AribaConnector implements Connector {
+  private static final Logger LOG = LoggerFactory.getLogger(AribaConnector.class);
+  private static final String METADATA_PATH = "api/analytics-reporting-view/v1";
+  private static final String PRODUCT = "product";
+  private static final String ANALYTICS = "analytics";
+  private static final String REALM = "realm";
+  private static final String ENTITY_TYPE_TEMPLATE = "template";
+  private static final Gson gson = new Gson();
+  private final AribaConnectorConfig config;
+  public String accessToken;
+
+  AribaConnector(AribaConnectorConfig config) {
+
+    this.config = config;
+  }
+
+  @Override
+  public void test(ConnectorContext connectorContext) throws ValidationException {
+    FailureCollector collector = connectorContext.getFailureCollector();
+    config.validateCredentials(collector);
+    try {
+      config.validateToken(collector);
+    } catch (AribaException | IOException e) {
+      LOG.error("Credentials are incorrect", e);
+    }
+  }
+
+  @Override
+  public BrowseDetail browse(ConnectorContext connectorContext, BrowseRequest browseRequest) throws IOException {
+    BrowseDetail.Builder browseDetailBuilder = BrowseDetail.builder();
+    int count = 0;
+    JsonArray jsonArray = listTemplates();
+    for (int i = 0; i < jsonArray.size(); i++) {
+      JsonObject resultJsonObject = gson.fromJson(jsonArray.get(i), JsonObject.class);
+      String name = resultJsonObject.get("viewTemplateName").getAsString();
+      BrowseEntity.Builder entity = (BrowseEntity.builder(name, name, ENTITY_TYPE_TEMPLATE).
+        canBrowse(false).canSample(true));
```

Comment:
cansample should be false if you are not implementing sample method. Make it false and test it.

Reply:
If it is declared false then user will not be able to select anything from the list.

Reply:
did you test this?